utils/filterPlayerProfileRead.ts

```diff
@@ -8,8 +8,16 @@ import {
 import ProfileAccessDefinitionsByRole from "lib/access/definitions";
 import resolveAccessValue from "lib/access/resolve";
 
-const filterPlayerProfileRead = (player: IPlayer, user: IUser): IPlayer => {
-  if (!player.profile || user.defaultRole.type === UserRoleType.Admin) {
+const filterPlayerProfileRead = (
+  player: IPlayer,
+  user: IUser,
+  isLinked?: boolean
+): IPlayer => {
+  if (
+    !player.profile ||
+    user.defaultRole.type === UserRoleType.Admin ||
+    isLinked
```

Comment:
You shouldn't need this extra parameter, as lib/access/definitions takes care of identifying whether or not the `player` is "linked" to the `user` (see `isSharedPlayerProfile`). In the cases where they are linked, it's not always true that we return the full profile as well!